components/supervisor/pkg/terminal/terminal.go

```diff
@@ -77,52 +78,54 @@ func (m *Mux) Start(cmd *exec.Cmd, options TermOptions) (alias string, err error
 	go func() {
 		term.waitErr = cmd.Wait()
 		close(term.waitDone)
-		_ = m.CloseTerminal(alias, 0*time.Second)
+		_ = m.CloseTerminal(alias)
 	}()
 
 	return alias, nil
 }
 
-// Close closes all terminals with closeTerminaldefaultGracePeriod.
+// Close closes all terminals.
 func (m *Mux) Close() error {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
+	var wg sync.WaitGroup
 	var err error
 	for k := range m.terms {
-		cerr := m.doClose(k, closeTerminaldefaultGracePeriod)
+		cerr := m.doClose(&wg, k)
```

Comment:
Still stubbing my toe on the waitgroup. Alternative:

```Go
var eg errgrp.Group
for k := range m.terms {
    // capture loop variable - DO NOT REMOVE THIS LINE
    k := k
    eg.Go(func() error {
        return m.doCLose(k)
    })
}
return eg.Wait()
```

Reply:
ah, I see you would keep the asynchronicity local.